dbms/src/Interpreters/Join.cpp

```diff
@@ -2451,6 +2756,15 @@ void Join::finishOneBuild()
 
 void Join::workAfterBuildFinish()
 {
+    if (isEnableSpill())
```

Comment:
The spill clean staff should be the last step of `workAfterBuildFinish`

Reply:
> The spill clean staff should be the last step of `workAfterBuildFinish`

done.